### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by aspateri**
Created Codeql yml to run every 5 mins

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '41 13 * * 4'
```
Comment (zendu) [inline .github/workflows/codeql.yml:21]: Schedule to run every week or every month as per CodeQL requirement.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
```
Comment (zendu) [inline .github/workflows/codeql.yml:31]: Simplify expression to use latest ubuntu.

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```
Comment (zendu) [inline .github/workflows/codeql.yml:32]: Simplify expression to set timeout to 1 hour.

**Review by zendu — CHANGES_REQUESTED** [review]

**Review by alberto-baron — APPROVED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (aspateri) [reply .github/workflows/codeql.yml:21]: This runs every week at 13:41 Thursday weekly

**Pull request merged**